src/main/scala/mesosphere/marathon/core/task/tracker/TaskStatusUpdateStep.scala

```diff
@@ -15,5 +15,5 @@ trait TaskStatusUpdateStep {
   def name: String
 
   def processUpdate(
-    timestamp: Timestamp, appId: PathId, maybeTask: Option[MarathonTask], mesosStatus: TaskStatus): Future[_]
+    timestamp: Timestamp, appId: PathId, task: MarathonTask, mesosStatus: TaskStatus): Future[_]
```

Comment:
Definitively an enhancement to not think about non existing tasks in every step!